src/block/block_read.c

```diff
@@ -197,12 +199,13 @@ __wt_block_read_off(WT_SESSION_IMPL *session, WT_BLOCK *block, WT_ITEM *buf, uin
     buf->size = size;
 
     /*
-     * Check if the chunk cache has the needed data. If it does not, the chunk cache may read it
-     * from the file.
+     * Check if the chunk cache has the needed data. If it does not, read it.
      */
     if (S2C(session)->chunkcache.configured)
-        WT_RET(__wt_chunkcache_get(session, block, objectid, offset, size, buf->mem));
-    else
+        WT_RET_ERROR_OK(
+          __wt_chunkcache_get(session, block, objectid, offset, size, buf->mem, &chunkcache_hit),
+          ENOSPC);
+    if (!chunkcache_hit)
```

Comment:
After some reading, I think I get this change. If the chunk cache is configured we try to get the data from it. If it comes back with `ENOSPC`, that's OK but it won't be hit. Therefore, if it's not hit, we need to fall back to a `__wt_read`.

Two things - just for my understanding, this is a bugfix, right? We previously would've done a `WT_RET` on the `ENOSPC` returned by `__chunkcache_alloc_chunk` and failed to do the read at all if there wasn't room in the chunk cache.

Secondly, this code makes sense, but in my opinion is a little hard to read now that it mixes the configuration check with a hit check, and you have to know that it doesn't (say) do the read behind your back if it can't find space. 

As such, can I ask for the comment here to elaborate on this a little more please? It's no longer quite the full story to say it doesn't have the data.

Reply:
@korteland, you understood this correctly, and yes, this is a bug fix. Will add the comment.